### ISSUE TITLE
on animation finish callback

### ISSUE THREAD

**Issue by xhy279** [konvajs/react-konva#535]
Im currently using react-konva in our project. I checked the docs and didn't get much info about animation finish callback in Konva.Animation(not Konva.Tween). Can anyone help on this?

**Comment by lavrton** [comment]
`Konva.Animation` doesn't have a finish callback. You need to finish it manually with `anim.stop()`

**Comment by xhy279** [comment]
> `Konva.Animation` doesn't have a finish callback. You need to finish it manually with `anim.stop()`

Thanks for your quick reply then my question is where to know the animation duration?  I suppose i may need a setTimeout using the duration if I want to achieve a similar functionality?

**Comment by xhy279** [comment]
Let me rephrase my question. Is there a way to cancel the current animation inside the animation loop? Thanks.

**Comment by lavrton** [comment]
There are many ways to stop animation. You can use `setTimeout` or just check `frame` argument of animation function:

```js
const anim = new Konva.Animation((frame) => {
  if (frame.time > 1000) {
    anim.stop();
  }
  shape.x(shape.x() + 1);
}, [layer]);
```

https://jsbin.com/lefesijahi/edit?js,output
